engine.py

```diff
@@ -3,306 +3,319 @@
+                    #	moves.append(Move((r, c), (i, j), (self.board))) # create a move object and append to moves
+                    if ((i == r+2) and (j == c) and (r==1) and (self.board[i][j] == "  ")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+                    if ((i == r+1) and (j == c) and (self.board[i][j] == "  ")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+                    if ((i == r+1) and (j == c+1 or j==c-1) and (self.board[i][j][1] == "l")):
+                        moves.append(Move((r, c), (i, j), self.board))
+
+
+    def get_bishop_moves(self, r, c, moves):
+
+         """
+            calculates all possible bishop moves for a given colour (light or dark)
+            and appends them to a list
+
+            input parameters:
+            r     --> starting row (int)
+            c     --> starting column (int)
+            moves --> posiible moves container (list)
+
+            return parameter(s):
+            None
+         """
+
+         ##TODO
+         pass
+
+
+    def get_knight_moves(self, r, c, moves):
+         ##TODO
+         pass
+
+    
+    def get_king_moves(self, r, c, moves):
+         ##TODO
+         pass
+
+
+
+    def get_rook_moves(self, r, c, moves):
+         direction = (-1, 1) # possible direction
+         
+         if self.light_to_move:
+            
+
+            # if it's light's turn to move
+            for d in direction:
+                #rows
+                for i in range(1,len(self.board)):
+                    rownum = r + d*i
+                    if (0 <= rownum < len(self.board)): # making sure row is on board
+                        if self.board[rownum][c] == "  ": # if square is empty
+                            moves.append(Move((r,c), (rownum,c), (self.board)))
+                        elif self.board[rownum][c][1] == "d": # if square has opponent piece
+                            moves.append(Move((r,c), (rownum,c), (self.board)))
+                            break
+                        else:
+                            break # when ally piece encountered
+                    else:
+                        break # when off the board
+                #columns
+                for i in range(1,len(self.board)):
+                    colnum = c + d*i
+                    if (0 <= colnum < len(self.board)): # making sure column is on board
+                        if self.board[r][colnum] == "  ": # if square is empty
+                            moves.append(Move((r,c), (r,colnum), (self.board)))
+                        elif self.board[r][colnum][1] == "d": # if square has opponent piece
+                            moves.append(Move((r,c), (r,colnum), (self.board)))
+                            break
+                        else:
+                            break # when ally piece encountered
+                    else:
+                        break # when off the board
```

Comment:
Add a few blank lines under this line to space out the white moves from the dark for readability

Reply:
All solved